purchase_invoice.py

```diff
@@ -1178,73 +378,73 @@ def store_json(json_data: dict):
+            purchase_invoice_google_parser = PurchaseInvoiceGoogleParser(self, invoice_json, given_supplier, is_test)
+            purchase_invoice_google_parser.set_purchase_info()
+            google_purchase_data = purchase_invoice_google_parser.get_purchase_data()
+        print("Nutze internen Parser")
+        self.extract_items = False
+        lines = pdf_to_text(infile)
         try:
-            self.extract_main_info(invoice_json, given_supplier, infile)
-            self.compute_total()
+            if lines:
+                head = lines[0][0:140]
+                if not head[0:10].split():
+                    for line in lines[0:10]:
+                        if self.company.name != 'Bremer SolidarStrom' and len(line) > 2 and (
+                                line[-2] == '£' or line[-3] == '£'):
+                            head = "Kornkraft Naturkost GmbH"
+                            break
+                supps = dict(PurchaseInvoice.suppliers)
+                if self.company.name != 'Laden':
+                    del supps['Rechnung']
+                for supplier, info in supps.items():
+                    if supplier in head:
+                        self.parser = supplier
+                        self.extract_items = self.update_stock
+                        if info['raw']:
+                            self.raw = True
+                            lines = pdf_to_text(infile, True)
+                        if 'supplier' in info:
+                            self.supplier = info['supplier']
+                        else:
+                            self.supplier = supplier
+                        print("Verwende Rechnungsparser für ", self.supplier)
+                        if given_supplier and self.supplier != given_supplier:
+                            print("abweichend von PreRechnung: ", given_supplier)
+                        self.multi = info['multi']
+                        purchase_invoice_parser = PurchaseInvoiceParser(self, info['parser'], lines)
+                        purchase_invoice_parser.set_purchase_info()
+                        normal_purchase_data = purchase_invoice_parser.get_purchase_data()
         except Exception as e:
-            if self.update_stock:
+            if google_purchase_data:
+                print(e)
+            elif self.update_stock:
                 raise e
             elif not is_test:
                 print(e)
                 print("Rückfall auf Standard-Rechnungsbehandlung")
+        if google_purchase_data:
+            if normal_purchase_data:
+                google_purchase_data.update(normal_purchase_data)
+                diff = jsondiff.diff(normal_purchase_data, google_purchase_data, syntax='symmetric')
+                self.apply_info_changes(diff, normal_purchase_data)
+            final_data = google_purchase_data
+            if manual_edit:
+                final_data = self.edit_data_model_manually(google_purchase_data, infile)
```

Comment:
the manual editing should be called in any case, also if there is no google_purchase_data.

Reply:
Done